File: Python/LeetCode/pascalsTriangle2.py
"""
Given an index k, return the kth row of the Pascal's triangle.

For example, given k = 3,
Return [1,3,3,1].

Note:
Could you optimize your algorithm to use only O(k) extra space?
"""

class Solution:
    # @return a list of integers
    def getRow(self, rowIndex):
        result = [0] * (rowIndex+1)
        for i in range(rowIndex+1):
            result[i] = 1
            if i > 1:
                for j in range(i-1, 0, -1):
                    if j > 0:
                        result[j] = result[j] + result[j-1]
        return result

solution = Solution()
print solution.getRow(4)
print solution.getRow(0)

File: CPP/LeetCode/pascalsTriangle2.cpp
/*
 * Given an index k, return the kth row of the Pascal's triangle.

For example, given k = 3,
Return [1,3,3,1].
*/

#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getRow(int rowIndex) {
      vector<int> result(1,1);
      if (rowIndex == 0)
        return result;
      for (int i = 1; i <= rowIndex; i++){
        result.push_back(1);
        for (int j = i-1; j >=1; j--){
          result[j] = result[j] + result [j-1];
        }
      }
      return result;
    }
};

int main(){
}
